openproblems/tasks/_batch_integration/batch_integration_graph/methods/_utils.py

```diff
@@ -19,7 +19,9 @@ def hvg_batch(adata, batch_key, target_genes, adataOut):
 def scale_batch(adata, batch_key):
     from scib.preprocessing import scale_batch
 
+    adata.var_names_make_unique()
+    adata.obs_names_make_unique()
     var = adata.var.copy()
     adata = scale_batch(adata, batch_key)
-    adata.var = var
+    adata.var = var.loc[adata.var_names]
```

Comment:
this is also making is more slim, right? Also for the dataset function, not in a method utils function.